WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingContainerListener.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.labkey.wnprc_purchasing;
+
+import org.jetbrains.annotations.NotNull;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.ContainerManager.ContainerListener;
+import org.labkey.api.security.User;
+import java.util.Collections;
+import java.util.Collection;
+
+import java.beans.PropertyChangeEvent;
+
+public class WNPRC_PurchasingContainerListener implements ContainerListener
+{
+    @Override
+    public void containerCreated(Container c, User user)
+    {
+        //populate qc state data upon WNPRC_Purchasing folder creation
+        WNPRC_PurchasingManager.get().addQCStatus(c, user);
+    }
```

Comment:
I believe this will apply to all newly created containers, regardless of folder type. Also, it wouldn't apply to existing folders that have their type changed.

Reply:
Good point Trey, I was thinking this was on a folder type but it's on the ContainerListener.

@labkey-bpatel I suggest introducing a FolderType implementation and doing this setup in its configureContainer() instead.

Reply:
Thank you for catching that, Trey! I missed that it would add data regardless of folder type.
Now updated!